src/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,164 @@
+import {
+	Component,
+	Input,
+	Output,
+	ViewChild,
+	EventEmitter,
+	OnInit
+} from "@angular/core";
+import { NG_VALUE_ACCESSOR } from "@angular/forms";
+
+import { I18n } from "../i18n/i18n.module";
+import { FileItem } from "./file-item.interface";
+
+const noop = () => {};
+
+@Component({
+	selector: "ibm-file-uploader",
+	template: `
+		<strong class="bx--label">{{title}}</strong>
+		<p class="bx--label-description">{{description}}</p>
+		<div class="bx--file">
+			<button
+				ibmButton="secondary"
+				(click)="fileInput.click()"
+				[attr.for]="fileUploaderId">
+				{{buttonText}}
+			</button>
+			<input
+				#fileInput
+				type="file"
+				class="bx--file-input"
+				[accept]="accept"
+				[id]="fileUploaderId"
+				[multiple]="multiple"
+				(change)="onFilesAdded()"/>
+			<div class="bx--file-container">
+				<ibm-file *ngFor="let fileItem of files" [fileItem]="fileItem" (remove)="removeFile(fileItem)"></ibm-file>
+			</div>
+		</div>
+	`,
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: FileUploader,
+			multi: true
+		}
+	]
+})
+export class FileUploader implements OnInit {
+	/**
+	 * Counter used to create unique ids for file-uploader components
+	 */
+	static fileUploaderCount = 0;
+	/**
+	 * Accessible text for the button that opens the upload window
```

Comment:
If there's no new line, text concatenates to the previous line.

```suggestion
	 * Accessible text for the button that opens the upload window.
	 *
```